crates/bevy_asset/src/path.rs

```diff
@@ -255,7 +288,7 @@ impl<'a> AssetPath<'a> {
     /// Gets the path to the asset in the "virtual filesystem".
     #[inline]
     pub fn path(&self) -> &Path {
```

Comment:
Should we be returning a str here instead? The comment says "path to the asset in the virtual filesystem", which implies we shouldn't be returning a native OS path.

Reply:
I think we should remove this in this PR. It will be more changes, but it will ensure we aren't accidentally converting to paths somewhere.